src/Features/CSharp/Portable/UseExpressionBodyForLambda/UseExpressionBodyForLambdaHelpers.cs

```diff
@@ -0,0 +1,227 @@
+        {
+            var currentOptionValue = optionSet.GetOption(CSharpCodeStyleOptions.PreferExpressionBodiedLambdas);
+            var preference = currentOptionValue.Value;
+            var userPrefersExpressionBodies = preference != ExpressionBodyPreference.Never;
+            var analyzerDisabled = currentOptionValue.Notification.Severity == ReportDiagnostic.Suppress;
+
+            // If the user likes expression bodies, then we offer expression bodies from the diagnostic analyzer.
+            // If the user does not like expression bodies then we offer expression bodies from the refactoring provider.
+            // If the analyzer is disabled completely, the refactoring is enabled in both directions.
+            if (userPrefersExpressionBodies == forAnalyzer || (!forAnalyzer && analyzerDisabled))
+            {
+                var expressionBody = GetExpressionBody(declaration);
+                if (expressionBody == null)
+                {
+                    // They don't have an expression body.  See if we could convert the block they 
+                    // have into one.
+
+                    var options = declaration.SyntaxTree.Options;
+                    var conversionPreference = forAnalyzer ? preference : ExpressionBodyPreference.WhenPossible;
+
+                    return TryConvertToExpressionBody(declaration, options, conversionPreference, out _, out _);
+                }
+            }
+
+            return false;
+        }
+
+        private static bool TryConvertToExpressionBody(
+            LambdaExpressionSyntax declaration,
+            ParseOptions options, ExpressionBodyPreference conversionPreference, 
+            out ExpressionSyntax expressionWhenOnSingleLine, 
+            out SyntaxToken semicolonWhenOnSingleLine)
+        {
+            return TryConvertToExpressionBodyWorker(
+                declaration, options, conversionPreference,
+                out expressionWhenOnSingleLine, out semicolonWhenOnSingleLine);
+        }
+
+        private static bool TryConvertToExpressionBodyWorker(
+            LambdaExpressionSyntax declaration, ParseOptions options, ExpressionBodyPreference conversionPreference,
+            out ExpressionSyntax expressionWhenOnSingleLine, out SyntaxToken semicolonWhenOnSingleLine)
+        {
+            var body = declaration.Body as BlockSyntax;
+
+            return body.TryConvertToExpressionBody(
+                declaration.Kind(), options, conversionPreference,
+                out expressionWhenOnSingleLine, out semicolonWhenOnSingleLine);
+        }
+
+        public static (bool canOffer, bool fixesError) CanOfferUseBlockBody(
+            SemanticModel semanticModel, OptionSet optionSet,
+            LambdaExpressionSyntax declaration, bool forAnalyzer,
+            CancellationToken cancellationToken)
+        {
+            var currentOptionValue = optionSet.GetOption(CSharpCodeStyleOptions.PreferExpressionBodiedLambdas);
```

Comment:
Those four locals can be declared closer to usage (last case in the chain).

Reply:
Done!